mlflow/lightgbm.py

```diff
@@ -669,6 +670,15 @@ def infer_model_signature(input_example):
 
         return model
 
+    def train(_log_models, original, *args, **kwargs):
+        topmost_sklearn_session = _SklearnTrainingSession.get_topmost_session()
+        if topmost_sklearn_session is None or (
+                topmost_sklearn_session is not None and topmost_sklearn_session.should_log()
```

Comment:
The test `tests/lightgbm/test_lightgbm_autolog.py::test_lgb_autolog_sklearn` failed because:

mlflow patches both `LGBMModel.fit` and `LGBMClassifier.fit`, and `LGBMClassifier.fit` calls `LGBMModel.fit`, the calling stack is like:

(1) patched LGBMClassifier.fit ->
(2) original LGBMClassifier.fit ->
(3) patched LGBMModel.fit ->
(4) original LGBModel.fit ->
(5) patched `lightgbm.train`
(5) original `lightgbm.train` 

The depth 5th call does the real metrics autologging.
and at the  depth 5th call place, we need to check whether current estimator is nested estimator, this is a bit hard because the training session stack is influenced by depth 3th call. 
So I create another PR to support this case:
https://github.com/mlflow/mlflow/pull/7411
with [that](https://github.com/mlflow/mlflow/pull/7411) PR, and changing `topmost_sklearn_session.should_log()` to be `topmost_sklearn_session.should_log_without_reentrance` can fix this issue.

Reply:
@harupy suggested a simpler fix:

> Inside lightgbm.train patching, we can just check the presence of sklearn.xxx class session in the stacks, if yes, then skip autologging.

This approach works for this case, and does not need this PR https://github.com/mlflow/mlflow/pull/7411 

On the other hand, this PR https://github.com/mlflow/mlflow/pull/7411 makes the session stack to be in a correct status, e.g. for sklearn estimators that calling `fit_predict` / `fit_transform` (they will trigger nested call on self estimator's fit), making the session stack in a correct status can helps us to extend functionality in future.

CC @dbczumar @ben Thoughts ?